src/lib/bootloader/grub2_widgets.rb

```diff
@@ -336,6 +338,19 @@ def init
     def store
       grub2.secure_boot = value
     end
+
+    def validate
+      return true if Yast::Mode.config ||
+        !Yast::Arch.s390 ||
+        value == Systeminfo.secure_boot_active?
+
+      Yast::Popup.ContinueCancel(
+        _(
+          "Make sure the Secure Boot setting matches the configuration of the HMC.\n\n" \
+          "Otherwise this system will not boot."
+        )
+      )
```

Comment:
One more usability note. There is method focus which is often used in validate to "highlight" given widget. So here it can be something like
```
res = Yast::Popup.ContinueCancel(...)
focus unless res
res
```

https://github.com/yast/yast-yast2/blob/master/library/cwm/src/lib/cwm/abstract_widget.rb#L187

Reply:
Again, this just mirrors the trusted boot dialog. Consistency across a UI is an important point. It doesn't help to do similar things in a completely different way.